src/main/python/deep_qa/layers/noisy_or.py

```diff
@@ -0,0 +1,88 @@
+from keras import backend as K
+from keras.layers import Layer
+from keras.initializations import identity
+from keras.constraints import Constraint, NonNeg
+from keras.regularizers import L1L2Regularizer
+from overrides import overrides
+
+class BetweenZeroAndOne(Constraint):
+    """Constrains the weights to be between zero and one
+    """
+    def __call__(self, p):
+        # handle negatives
+        p *= K.cast(p >= 0., K.floatx())
+        leaky_zeros_mask = K.epsilon() * K.cast(K.equal(p, K.zeros_like(p)), 'float32')
+        p = p + leaky_zeros_mask
+        # handle greater than 1
+        p *= K.cast(p <= 1., K.floatx())
+        leaky_ones_mask = 1.0 * K.cast(K.equal(p, K.zeros_like(p)), K.floatx())
+        p += leaky_ones_mask
+        return p
+
+
+class NoisyOr(Layer):
+
+    def __init__(self, axis=-1, param_init='uniform', prefix="noisy_or", **kwargs):
+        self.axis = axis
+        self.param_init = param_init
+        self.supports_masking = True
+        self.prefix = prefix
+        super(NoisyOr, self).__init__(**kwargs)
+
+    def build(self, input_shape):
+        # Create a trainable weight variable for the noise parameter
+        self.noise_parameter = self.add_weight(shape=(1,),
+                                               name=self.prefix + '_q',
+                                               initializer=self.param_init,
+                                               regularizer=L1L2Regularizer(l2=0.001),
+                                               # constraint=BetweenZeroAndOne(),
+                                 trainable=True)
+        super(NoisyOr, self).build(input_shape)
+
+    def get_output_shape_for(self, input_shape):
+        if self.axis == -1:
+            return input_shape[0:-1]
+        return input_shape[:self.axis - 1] + input_shape[self.axis:]
+
+    @overrides
+    def compute_mask(self, inputs, mask=None):
+        # pylint: disable=unused-argument
+        # We do not need a mask beyond this layer.
+        # (probably right)
+        if mask is not None:
+            return K.any(mask, axis=-1)
+        return None
+
+    def call(self, x, mask=None):
+        """
```

Comment:
I think this docstring is a lot more useful on the class, instead of the `call()` method.  You could duplicate it here, if you want, but this information needs to be in a class docstring.  For a good example, see here: https://github.com/allenai/deep_qa/blob/master/src/main/python/deep_qa/layers/attention/gated_attention.py.